streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -558,7 +559,7 @@ void commit(final boolean startNewTransaction, final Map<TopicPartition, Long> p
             } else {
                 consumer.commitSync(consumedOffsetsAndMetadata);
             }
-        } catch (final CommitFailedException | ProducerFencedException error) {
+        } catch (final CommitFailedException | ProducerFencedException | UnknownProducerIdException error) {
```

Comment:
Ok, I wasn't sure, so I just put it everywhere that we already check for `ProducerFencedException`, on the rationale that, if we can get fenced, then we must have a transactional id, and if we have an id, then it could be "unknown". Clearly, this is more intuitive than analytical, so I find your feedback plausible.